internal/events/events.go

```diff
@@ -126,14 +126,25 @@ func eventString(writer io.Writer, event flow.Event) {
 }
 
 func printField(writer io.Writer, field cadence.Field, value cadence.Value) {
+	v := value.String()
 	var typeId string
+
+	defer func() {
+		if err := recover(); err != nil {
+			v = fmt.Sprintf("%s\n\t\thex: %x", v, v)
+			_, _ = fmt.Fprintf(writer, "\t\t- %s (%s): %s \n", field.Identifier, "?", v)
+		}
+	}()
+
 	if field.Type != nil {
+		//TODO: onflow/cadence issue #1672
+		//currently getting ID for cadence array will cause panic
 		typeId = field.Type.ID()
 	}
 
-	v := value.String()
 	if typeId == "" { // exception for not known typeId workaround for cadence arrays
 		v = fmt.Sprintf("%s\n\t\thex: %x", v, v)
```

Comment:
why do we output value as hex here?

Reply:
The hex was already there line 146 and below, which is triggered for array types like [String] (and not [String]?) Should we remove the hex entirely for both?

Reply:
I see. You are right. Lets just keep it as it was before but only for not known type ids